src/App.jsx

```diff
@@ -14,33 +19,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [sortMethod, setSortMethod] = useState('');
+  const [reverse, setReverse] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const copyOfGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const SORT_ALPHABETICALY = 'alphabet';
+  const SORT_BY_LENGTH = 'length';
+  const REVERSED_SORT = 'reversed';
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const resetFunction = () => {
+    setReverse('');
+    setSortMethod('');
+  };
+
+  const checkForSecondClick = (current, setMethod, sortParameter) => {
+    if (sortParameter === current) {
+      setMethod('');
+    } else {
+      setMethod(sortParameter);
+    }
+  };
+
+  const changeList = () => {
+    switch (sortMethod) {
+      case SORT_ALPHABETICALY:
+        copyOfGoods.sort((obj1, obj2) => obj1.localeCompare(obj2));
+        break;
+
+      case SORT_BY_LENGTH:
+        copyOfGoods.sort((obj1, obj2) => obj1.length - obj2.length);
+        break;
+    }
+
+    if (reverse === REVERSED_SORT) {
+      copyOfGoods.reverse();
+    }
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  changeList();
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': SORT_ALPHABETICALY !== sortMethod,
+          })}
+          onClick={() => {
+            checkForSecondClick(sortMethod, setSortMethod, SORT_ALPHABETICALY);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': SORT_BY_LENGTH !== sortMethod,
+          })}
+          onClick={() => {
+            checkForSecondClick(sortMethod, setSortMethod, SORT_BY_LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': REVERSED_SORT !== reverse,
+          })}
+          onClick={() => {
+            checkForSecondClick(reverse, setReverse, REVERSED_SORT);
```

Comment:
looks too complicated
```suggestion
            setReverse(!reverse);
```